<a href="https://colab.research.google.com/github/Sergius-DS/elecciOn_datasets/blob/main/Dataset_Reducido.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# DATA SET DIABETES

## FUNCIONES

Crear funciones para luego llamarlas desde un archivo aparte como cuando uno llama import pandas o from sklearn import (algo), así se ve más estético el notebook en Colab

In [ ]:
%%writefile funciones.py


Overwriting funciones.py


# INICIO

In [2]:
import pandas as pd
import numpy as np
import warnings
warnings.filterwarnings('ignore')
# llamar las funciones creadas también se puede from funciones import *
#from funciones import categorizar_columnas, verifica_consistencia_filas
#from funciones import *
pd.set_option('display.max_columns', None)
pd.set_option('display.float_format', '{:.2f}'.format)

In [11]:
ds = pd.read_csv('data_diabetes.csv')

In [12]:
ds.shape

(867970, 18)

In [13]:
ds.head(4)

FECHA_AFILIADOS_SIS                                 CODIGO_ANONIMIZADO  \
0          2023-03-31  0C10D14E8C401B6B5B379793BCBE586F4697366ECE8D0A...   
1          2023-03-31  CAEA7585BB5B1A1B8CB6A8612931EE04EDFF258BE91E65...   
2          2023-03-31  19CDB3BC20FA3B6FDB458049C73BBDA679E9E3AD54D133...   
3          2023-09-30  C64591A0098FC4539D342C02F3D0EF0918D5A09E63241B...   

   EDAD DEPARTAMENTO   PROVINCIA              DISTRITO       SEXO  \
0    54        CUSCO       CUSCO         SAN SEBASTIAN  MASCULINO   
1    40         LIMA       CANTA  SANTA ROSA DE QUIVES   FEMENINO   
2    62          ICA     CHINCHA          CHINCHA BAJA   FEMENINO   
3    65   LAMBAYEQUE  LAMBAYEQUE               MOCHUMI   FEMENINO   

  FECHA_PRIMER_DX             TIPO_DIABETES  CON_DX_OBESIDAD  \
0      2019-01-11  Diabetes mellitus tipo 2                0   
1      2019-02-22  Diabetes mellitus tipo 2                0   
2      2020-08-04  Diabetes mellitus tipo 2                0   
3      2018-07-09  Diabetes mellitus tipo 2                1   

   CON_DX_HIPERTENSION  CON_DX_SALUDMENTAL  CANT_ATENCIONES  VALOR_NETO  \
0                    0                   0                0        0.00   
1                    0                   0                0        0.00   
2                    1                   1                0        0.00   
3                    1                   0                0        0.00   

   CANT_ATENCIONES_HOSP  VALOR_NETO_HOSP  DIAS_HOSP  HOSPITALIZADO  
0                     0             0.00          0              0  
1                     0             0.00          0              0  
2                     0             0.00          0              0  
3                     0             0.00          0              0

In [19]:
ds[ds['FECHA_AFILIADOS_SIS'] == '2023-06-30']

FECHA_AFILIADOS_SIS                                 CODIGO_ANONIMIZADO  \
4               2023-06-30  03ADFFBAD7BFDB90E9E5293E3B9DC33406A8203278E73A...   
5               2023-06-30  EB9DC5E84F4E21E9D3142BB9DC683E4756636CF6947554...   
9               2023-06-30  3AB2DCC4C811699886EF17DFB3B7C1FBE430BD98385014...   
11              2023-06-30  60816D801AEF0B604CFFC3540FBA5090C6C87F1AF6E6D2...   
14              2023-06-30  D7830949C80B4C944727EB2A41F6903EFF6F1F87C4AAA4...   
...                    ...                                                ...   
867957          2023-06-30  D35536DD4C22C72E3813F18260AD81EB7AE0779F0617A1...   
867962          2023-06-30  0BDF3D58FEF96F57B4B35330F3ED29A69A4A9D7443ECDF...   
867963          2023-06-30  0CE26D6D19974BCA8496245C6858A6782CF71E4C4BB423...   
867966          2023-06-30  2BA7413E1355E5BF9B5757DB3F06F7DE592AF6668B46A3...   
867967          2023-06-30  5B98C0A8683D9EFDA5FB77E41DD0142FCE2918CA0B210F...   

        EDAD  DEPARTAMENTO                PROVINCIA               DISTRITO  \
4         71          LIMA                     LIMA                   LIMA   
5         52         JUNÍN              CHANCHAMAYO                 PERENE   
9         38         CUSCO                    CUSCO           SAN JERÓNIMO   
11        58           ICA                    PISCO           SAN CLEMENTE   
14        49        CALLAO  PROV. CONST. DEL CALLAO                 CALLAO   
...      ...           ...                      ...                    ...   
867957    31          LIMA                     LIMA   SAN MARTÍN DE PORRES   
867962    66  HUANCAVELICA                 ANGARAES                 LIRCAY   
867963    76         PIURA                    PIURA  VEINTISEIS DE OCTUBRE   
867966    62         PIURA                    PAITA                  COLAN   
867967    37        TUMBES                ZARUMILLA           AGUAS VERDES   

             SEXO FECHA_PRIMER_DX                       TIPO_DIABETES  \
4        FEMENINO      2019-11-16            Diabetes mellitus tipo 2   
5        FEMENINO      2022-12-02            Diabetes mellitus tipo 2   
9        FEMENINO      2022-10-06            Diabetes mellitus tipo 2   
11       FEMENINO      2019-10-23            Diabetes mellitus tipo 2   
14       FEMENINO      2018-05-15            Diabetes mellitus tipo 2   
...           ...             ...                                 ...   
867957   FEMENINO      2023-03-03            Diabetes mellitus tipo 2   
867962   FEMENINO      2021-03-09            Diabetes mellitus tipo 2   
867963   FEMENINO      2022-06-03  Diabetes mellitus  no especificada   
867966  MASCULINO      2018-01-04            Diabetes mellitus tipo 2   
867967  MASCULINO      2020-01-20            Diabetes mellitus tipo 2   

        CON_DX_OBESIDAD  CON_DX_HIPERTENSION  CON_DX_SALUDMENTAL  \
4                     1                    1                   1   
5                     1                    0                   0   
9                     1                    0                   0   
11                    1                    0                   1   
14                    0                    0                   1   
...                 ...                  ...                 ...   
867957                1                    0                   1   
867962                0                    1                   1   
867963                1                    1                   0   
867966                1                    1                   0   
867967                1                    0                   0   

        CANT_ATENCIONES  VALOR_NETO  CANT_ATENCIONES_HOSP  VALOR_NETO_HOSP  \
4                     1        6.96                     0             0.00   
5                     1        2.04                     0             0.00   
9                     0        0.00                     0             0.00   
11                    0        0.00                     0             0.00   
14           

In [21]:
ds[ds[['CODIGO_ANONIMIZADO',	'EDAD','DEPARTAMENTO','PROVINCIA']].duplicated()]

FECHA_AFILIADOS_SIS                                 CODIGO_ANONIMIZADO  \
2533            2023-06-30  718E7957E0184E9C421EC9719320B670B25A0088D8962E...   
4224            2023-09-30  D6267A6CD82676D017CB45F4415D8B5B07EF12274721D6...   
4406            2023-09-30  B35750F2BBEF8C75C230DEC4E65E9623712E5152120C2F...   
4785            2023-06-30  CD511B03C815CF0D61A8AA9C14250B99D4911F6F34904A...   
5156            2023-06-30  C08CC81BAE0962C81C958560A2F88BECD78965148A6AA8...   
...                    ...                                                ...   
867958          2023-09-30  1A838774100542FE4EC2B38653246911951B112538D8B9...   
867964          2023-09-30  4DCEF3FDB37EEB488BE220D571C8C2B3D5FFC794F2F9C8...   
867966          2023-06-30  2BA7413E1355E5BF9B5757DB3F06F7DE592AF6668B46A3...   
867967          2023-06-30  5B98C0A8683D9EFDA5FB77E41DD0142FCE2918CA0B210F...   
867969          2023-09-30  816161EA5FE3D8878D6A6D68C52C9691677B572F840D16...   

        EDAD DEPARTAMENTO         PROVINCIA                DISTRITO  \
2533      55      UCAYALI  CORONEL PORTILLO             YARINACOCHA   
4224      60        CUSCO           ACOMAYO                 ACOMAYO   
4406      70         LIMA              LIMA              SAN MIGUEL   
4785      51         LIMA              LIMA           INDEPENDENCIA   
5156      69        PIURA          MORROPÓN              CHULUCANAS   
...      ...          ...               ...                     ...   
867958    53         LIMA              LIMA  SAN JUAN DE MIRAFLORES   
867964    79         PUNO            SANDIA                   PHARA   
867966    62        PIURA             PAITA                   COLAN   
867967    37       TUMBES         ZARUMILLA            AGUAS VERDES   
867969    66        PIURA          MORROPÓN              LA MATANZA   

             SEXO FECHA_PRIMER_DX                       TIPO_DIABETES  \
2533     FEMENINO      2019-08-01  Diabetes mellitus  no especificada   
4224     FEMENINO      2023-06-25            Diabetes mellitus tipo 2   
4406    MASCULINO      2019-02-22            Diabetes mellitus tipo 2   
4785     FEMENINO      2018-07-23            Diabetes mellitus tipo 2   
5156     FEMENINO      2021-02-03            Diabetes mellitus tipo 2   
...           ...             ...                                 ...   
867958  MASCULINO      2021-05-13  Diabetes mellitus  no especificada   
867964   FEMENINO      2020-01-21            Diabetes mellitus tipo 1   
867966  MASCULINO      2018-01-04            Diabetes mellitus tipo 2   
867967  MASCULINO      2020-01-20            Diabetes mellitus tipo 2   
867969   FEMENINO      2018-06-25            Diabetes mellitus tipo 2   

        CON_DX_OBESIDAD  CON_DX_HIPERTENSION  CON_DX_SALUDMENTAL  \
2533                  1                    0                   0   
4224                  0                    0                   0   
4406                  0                    1                   0   
4785                  1                    0                   0   
5156                  1                    0                   0   
...                 ...                  ...                 ...   
867958                0                    0                   0   
867964                1                    0                   0   
867966                1                    1                   0   
867967                1                    0                   0   
867969                1                    0                   0   

        CANT_ATENCIONES  VALOR_NETO  CANT_ATENCIONES_HOSP  VALOR_NETO_HOSP  \
2533                  2        0.00                     0             0.00   
4224                  0        0.00                     0             0.00   
4406                  0        0.00                     0             0.00   
4785                  0        0.00                     0             0.00   
5156                  0        0.00                     0             0.00   
...                

In [26]:
print(ds[ds['CODIGO_ANONIMIZADO'].str.startswith('718E7957E0184E9C421EC9719320B670')])

       FECHA_AFILIADOS_SIS                                 CODIGO_ANONIMIZADO  \
1548            2023-09-30  718E7957E0184E9C421EC9719320B670B25A0088D8962E...   
2533            2023-06-30  718E7957E0184E9C421EC9719320B670B25A0088D8962E...   
495848          2023-03-31  718E7957E0184E9C421EC9719320B670B25A0088D8962E...   

        EDAD DEPARTAMENTO         PROVINCIA     DISTRITO      SEXO  \
1548      55      UCAYALI  CORONEL PORTILLO  YARINACOCHA  FEMENINO   
2533      55      UCAYALI  CORONEL PORTILLO  YARINACOCHA  FEMENINO   
495848    55      UCAYALI  CORONEL PORTILLO  YARINACOCHA  FEMENINO   

       FECHA_PRIMER_DX                       TIPO_DIABETES  CON_DX_OBESIDAD  \
1548        2019-08-01  Diabetes mellitus  no especificada                1   
2533        2019-08-01  Diabetes mellitus  no especificada                1   
495848      2019-08-01  Diabetes mellitus  no especificada                1   

        CON_DX_HIPERTENSION  CON_DX_SALUDMENTAL  CANT_ATENCIONES  VALOR_NETO 

OBSERVACIONES:

Traté de buscar un código anonimizado al azar, para revisar o tratar de enter el porqué de las ocurrencias

In [35]:
aggregado_ds = ds.groupby('CODIGO_ANONIMIZADO').agg({
    'CANT_ATENCIONES': 'sum',
    'VALOR_NETO': 'sum',
    'CANT_ATENCIONES_HOSP': 'sum',
    'VALOR_NETO_HOSP': 'sum',
    'DIAS_HOSP': 'sum',
    'HOSPITALIZADO': 'max',
    'EDAD': 'first',  # Tomar la primera ocurrencia del data set
    'DEPARTAMENTO': 'first',
    'PROVINCIA': 'first',
    'DISTRITO': 'first',
    'SEXO': 'first',
    'FECHA_PRIMER_DX': 'first',
    'TIPO_DIABETES': 'first',
    'CON_DX_OBESIDAD': 'first',
    'CON_DX_HIPERTENSION': 'first',
    'CON_DX_SALUDMENTAL': 'first'
}).reset_index()


OBSERVACIONES:

De acuerdo a eso decidí sumarizar los datos, las atenciones sumarlas, el valor de costo neto, etc

In [37]:
aggregado_ds[aggregado_ds.duplicated()].sum()

CODIGO_ANONIMIZADO        0
CANT_ATENCIONES           0
VALOR_NETO             0.00
CANT_ATENCIONES_HOSP      0
VALOR_NETO_HOSP        0.00
DIAS_HOSP                 0
HOSPITALIZADO             0
EDAD                      0
DEPARTAMENTO              0
PROVINCIA                 0
DISTRITO                  0
SEXO                      0
FECHA_PRIMER_DX           0
TIPO_DIABETES             0
CON_DX_OBESIDAD           0
CON_DX_HIPERTENSION       0
CON_DX_SALUDMENTAL        0
dtype: object

In [38]:
aggregado_ds['HOSPITALIZADO'].value_counts(normalize=True)

HOSPITALIZADO
0   0.98
1   0.02
Name: proportion, dtype: float64

OBSERVACIONES:

Por lo que se ve se mantendría la misma proporción quizás, de bajar fue uno que otro número luego de la coma

In [39]:
aggregado_ds.describe()

CANT_ATENCIONES  VALOR_NETO  CANT_ATENCIONES_HOSP  VALOR_NETO_HOSP  \
count        523798.00   523798.00             523798.00        523798.00   
mean              0.92       87.43                  0.03            67.37   
std               1.87     1100.96                  0.21          1085.72   
min               0.00        0.00                  0.00             0.00   
25%               0.00        0.00                  0.00             0.00   
50%               0.00        0.00                  0.00             0.00   
75%               1.00        1.53                  0.00             0.00   
max             254.00   232695.27                 17.00        232695.27   

       DIAS_HOSP  HOSPITALIZADO      EDAD  CON_DX_OBESIDAD  \
count  523798.00      523798.00 523798.00        523798.00   
mean        0.30           0.02     55.63             0.63   
std         3.34           0.15     16.87             0.48   
min         0.00           0.00      0.00             0.00   
25%         0.00           0.00     46.00             0.00   
50%         0.00           0.00     57.00             1.00   
75%         0.00           0.00     67.00             1.00   
max       372.00           1.00    119.00             1.00   

       CON_DX_HIPERTENSION  CON_DX_SALUDMENTAL  
count            523798.00           523798.00  
mean                  0.37                0.22  
std                   0.48                0.41  
min                   0.00                0.00  
25%                   0.00                0.00  
50%                   0.00                0.00  
75%                   1.00                0.00  
max                   1.00                1.00

In [42]:
aggregado_ds[aggregado_ds['EDAD'] == 0].count()

CODIGO_ANONIMIZADO      117
CANT_ATENCIONES         117
VALOR_NETO              117
CANT_ATENCIONES_HOSP    117
VALOR_NETO_HOSP         117
DIAS_HOSP               117
HOSPITALIZADO           117
EDAD                    117
DEPARTAMENTO            117
PROVINCIA               117
DISTRITO                117
SEXO                    117
FECHA_PRIMER_DX         117
TIPO_DIABETES           117
CON_DX_OBESIDAD         117
CON_DX_HIPERTENSION     117
CON_DX_SALUDMENTAL      117
dtype: int64

OBSERVACIONES:

Tenemos 117 casos con pacientes que tienen edad CERO, no sé si quieren conservar "0" en sí o cambiar a "1", "CERO, ALGO" = "0.5" para que no vaya a afectar tanto al sacar el promedio

In [49]:
aggregado_ds[aggregado_ds['EDAD'] > 100].count()

CODIGO_ANONIMIZADO      95
CANT_ATENCIONES         95
VALOR_NETO              95
CANT_ATENCIONES_HOSP    95
VALOR_NETO_HOSP         95
DIAS_HOSP               95
HOSPITALIZADO           95
EDAD                    95
DEPARTAMENTO            95
PROVINCIA               95
DISTRITO                95
SEXO                    95
FECHA_PRIMER_DX         95
TIPO_DIABETES           95
CON_DX_OBESIDAD         95
CON_DX_HIPERTENSION     95
CON_DX_SALUDMENTAL      95
dtype: int64

OBSERVACIONES:

Aquí elegí de forma arbitaria una edad razonable hasta 100 años, yo creo máximo hasta 110, para ver qué opinan ustedes